File: django_apollo/blog/types/post_type.py
import graphene
from graphene.contrib.django import DjangoObjectType
from graphene.contrib.django.filter import DjangoFilterConnectionField
from blog.models import Post


class PostType(graphene.ObjectType):
    id = graphene.String()
    title = graphene.String(description='title of the post')
    description = graphene.String(description='description of the post')
    comments = graphene.List('commenttype', description='Comments of the post')
    user = graphene.Field(
        'usertype',
        id=graphene.ID(),
        description='User for this post',
    )

File: ruby/rails_graphql/app/api/types/post_type.rb
PostType = GraphQL::ObjectType.define do
  name "Post"
  description "A post"

  interfaces [NodeIdentification.interface]
  global_id_field :id

  field :title, !types.String
  field :body, !types.String
  field :user, -> { UserType }
  field :comments do
    type types[!CommentType]
    resolve -> (obj, args, ctx) { obj.comments.includes(:user) }
  end
end
